05/app.js

```diff
@@ -0,0 +1,51 @@
+function Student(initFirstName, initLastName) {
+  this.firstName = initFirstName;
+  this.lastName = initLastName;
+  this.grades = {};
+}
+
+Student.prototype.addGrade = function (subject, grade) {
+  if (!this.grades[subject]) {
+    this.grades[subject] = [];
+  }
+  this.grades[subject].push(grade);
+};
+
+Student.prototype.getAverageGrade = function (grade) {
+  if (
+    typeof grade !== 'undefined' &&
+    typeof this.grades[grade] !== 'undefined'
+  ) {
+    const grades = this.grades[grade];
+    if (grades.length === 0) {
+      throw new Error(`Brak ocen z przedmiotu ${subject}`);
+    }
+    const sum = grades.reduce((accumulator, current) => accumulator + current);
+    return sum / grades.length;
+  } else {
+    return 0;
```

Comment:
Widzę pewną nie konsekwencję przy "błędach" używasz `throw` natomiast tutaj nie. 
Myślę, że to podobna sytuacja i warto konsekwentnie zarządzać "wyjątkami" :)